pkg/agent/openflow/pipeline.go

```diff
@@ -533,7 +533,7 @@ func (f *featurePodConnectivity) gatewayClassifierFlows() []binding.Flow {
 			MatchInPort(f.gatewayPort).
 			MatchProtocol(ipProtocol).
 			MatchSrcIP(gatewayIP).
-			Action().LoadRegMark(FromGatewayRegMark).
```

Comment:
I am a bit confused, do you mean that Pod traffic in NetworkPolicyOnly scenario will use antrea gatewayIP as the source? If so, this is an invalid assumption.

Reply:
This is used to accommodate with `proxyAll` enabled. When `proxyAll` is enabled, the source IP of Service traffic from the local Node is the gateway IP, and such traffic should be also recognized as local. For NetworkPolicyOnly scenario, this flow will never be used as the Service traffic originated from the local Node will never be handled by Antrea.

Reply:
> What would prevent that?

So far, nothing prevents that, and antrea-agent gets crashed when `proxyAll` is enabled in networkPolicyMode.


> does that mean that proxyAll can never be used in policyOnly mode?

What I said about `this flow will never be used as the Service traffic originated from the local Node will never be handled by Antrea` was not rigorous. In fact. the `proxyAll` can be enabled in networkPolicyMode, but Antrea gets something wrong when setting static neighbor, which is prepared for installing static routes for forwarding Service traffic from the external network or the local Node to `antrea-gw0`.  

I have two simple ideas:

1. Prevent `proxyAll` being enabled in networkPolicyMode mode.
2. Fix the crash issue mentioned above, and `proxyAll` can be enabled in networkPolicyMode mode. However, I'm not sure if the route installed by Antrea for forwarding Service traffic has some "conflicts" with the primary CNI.

Reply:
Sure, will do it.